packages/ionio/src/Transaction.ts

```diff
@@ -280,17 +322,31 @@ export class Transaction implements TransactionInterface {
     }
 
     // check for blinding to be made
-    if (this.inputBlindingData.size > 0) {
-      if (this.outputBlindingPubKeys.size === 0)
+    if (this.inputBlinders.length > 0) {
+      const needsBlinding = this.pset.outputs.some((o: PsetOutput) =>
+        o.needsBlinding()
+      );
+      if (!needsBlinding)
         throw new Error(
           'if one confidential input is spent, at least one of the outputs must be blinded'
         );
 
-      await this.psbt.blindOutputsByIndex(
-        Psbt.ECCKeysGenerator(this.ecclib),
-        this.inputBlindingData,
-        this.outputBlindingPubKeys
+      const zkpValidator = new ZKPValidator();
+      const zkpGenerator = ZKPGenerator.fromOwnedInputs(this.inputBlinders);
```

Comment:
Question: `this.inputBlinders` is the list of unblinded prevouts? the naming is a bit misleading